server/src/main/java/com/epam/aidial/core/server/controller/ControllerSelector.java

```diff
@@ -405,4 +356,22 @@ private String resourcePath(String url) {
 
         return url.substring(prefix.length());
     }
-}
\ No newline at end of file
+
+    private record ControllerRoute(HttpMethod method, Pattern pathPattern, Initializer initializer) {
+        public ControllerSelection select(HttpMethod method, String path) {
+            if (method == null || this.method.equals(method)) {
```

Comment:
`if (method == null || this.method.equals(method))` -> `if (this.method.equals(method))`

Reply:
fixed